src-ui/src/app/components/document-detail/document-detail.component.ts

```diff
@@ -111,12 +117,29 @@ export class DocumentDetailComponent implements OnInit {
       this.suggestions = null
       if (this.openDocumentService.getOpenDocument(this.documentId)) {
         this.updateComponent(this.openDocumentService.getOpenDocument(this.documentId))
-      } else {
-        this.documentsService.get(this.documentId).subscribe(doc => {
+      }
+      this.documentsService.get(this.documentId).subscribe(doc => {
+        // Initialize dirtyCheck
+        this.store = new BehaviorSubject({
+          title: doc.title,
+          content: doc.content,
+          created: doc.created,
+          correspondent: doc.correspondent,
+          document_type: doc.document_type,
+          archive_serial_number: doc.archive_serial_number,
+          tags: doc.tags
+        })
+
+        this.isDirty$ = dirtyCheck(this.documentForm, this.store.asObservable())
+        this.isDirty$.subscribe(dirty => {
```

Comment:
pls avoid nested subscriptions. https://www.thinktecture.com/en/angular/rxjs-antipattern-1-nested-subs/

Reply:
Well this was something new for me, not incredibly facile with rxjs. Also there are other instances of this in the codebase but no reason to deal with here.

Reply:
Yes, I've seen this too, but from my point of view, there is no need to go through the complete codebase. Step by Step improvements are enough. If smb. come across on such a thing it can be fixed/improved.